command/events.go

```diff
@@ -147,19 +148,24 @@ func (c *EventsSubscribeCommands) subscribeRequest(client *api.Client, path stri
 			HTTPClient: client.CloneConfig().HttpClient,
 			HTTPHeader: client.Headers(),
 		})
-		if err != nil {
-			if resp != nil {
-				if resp.StatusCode == http.StatusNotFound {
-					return fmt.Errorf("events endpoint not found; check `vault read sys/experiments` to see if an events experiment is available but disabled")
-				} else if resp.StatusCode == http.StatusTemporaryRedirect {
-					url = resp.Header.Get("Location")
-					continue
-				}
-			}
+
+		if err == nil {
+			break
+		}
+
+		switch {
+		case resp == nil:
+			return err
```

Comment:
This and the default case look like behaviour changes to me. Previously we would retry in those cases, but now we give up immediately and return `err` which we know is `nil` from the `if err == nil { break }` check above.

Reply:
I think we only ever retried (up to an additional 9 times) previously if we received a `302` redirect and `continue`d. So this code seemed to be saying: Follow up to 9 redirects after the initial request, not 'retry requests if there are problems'.

Reply:
This is slightly different behavior, but I think it is still in-line with what I intended -- just to follow redirects.

Reply:
I put this together to try and help me make sure I thought we were doing the same stuff:
![image](https://github.com/hashicorp/vault/assets/487783/2cf72447-9f7c-4518-b80d-5ccf004f7baa)

I still feel like it's the same, I hope I didn't miss something, but if it's OK with @swenson as-is then that's cool.

Reply:
Looking at Pete's screenshot, the behavior looks the same to me. I will admit I had to read through it several times to get to that conclusion though 😁

Reply:
Likewise, it took me a lot of brainpower to verify 😄 thanks for the screenshot 👍